### PR TITLE
Add modify user backend logic

### LINKED ISSUE #113: Missing API endpoint for updating a user
## Description:
I would suggest the following endpoint to update username, email and password: PUT `/api/user` 

## Next steps:
- #104

### PR DESCRIPTION
**Pull request by TomRomeo**
## Description
This PR implements the backend logic to modify user data.

## Endpoint
`PUT` `/api/user`
```json
{
  "userName": "",
  "userEmail": "",
  "userPassword":""
}
```
## Resolved Issues
resolves #113

### CODE REVIEW

**Review by memeToasty — APPROVED** [review]

File: backend/services/frontend/handler/frontend.go [hunk 1]
```diff
@@ -192,6 +143,35 @@ func (e *Frontend) GetUserHandler(c *fiber.Ctx) error {
...
+	var data map[string]*string
+	if err := c.BodyParser(&data); err != nil {
+		return err
+	}
+	rspModUser, err := e.userService.ModifyUserProfileInformation(c.Context(), &pbUser.ModifyRequest{
+		UserID:       userID,
+		UserEmail:    data["userEmail"],
+		UserName:     data["userName"],
+		UserPassword: data["userPassword"],
+	})
```
Comment (darmiel) [inline backend/services/frontend/handler/frontend.go:157]: use a struct

```go
type modifyUserPayload struct {
  UserEmail string `json:"userEmail"`
  UserName string `json:"userName"`
  UserPassword string `json:"userPassword"`
}
// ...
var data modifyUserPayload
err := c.BodyParser(&data)
// ...
```
```diff
- UserEmail: data["userEmail"],
+ UserEmail: data.UserEmail,
```

File: backend/services/user/handler/user.go [hunk 2]
```diff
@@ -173,3 +181,40 @@ func (e *User) GetUserProfileInformation(
...
+	if req.UserPassword != nil {
+		hash, err := bcrypt.GenerateFromPassword([]byte(*req.UserPassword), bcrypt.MinCost)
+		if err != nil {
+			return helper.NewMicroHashingFailedErr(helper.UserServiceID)
+		}
+		user.Password = string(hash)
+	}
+	err = e.store.ModifyUser(user)
+	logger.Infof("Modified profile information for user with id %s", req.UserID)
+	rsp.Success = true
```
Comment (darmiel) [inline backend/services/user/handler/user.go:218]: You should probably handle this error (and not return `rsp.Success = true` if there is an error)

**Review by 4KevR — APPROVED** [review]
No other comments

File: backend/services/frontend/handler/frontend.go (hunk 1 above)
Comment (TomRomeo) [reply backend/services/frontend/handler/frontend.go]: implemented in 8344962

File: backend/services/user/handler/user.go (hunk 2 above)
Comment (TomRomeo) [reply backend/services/user/handler/user.go]: implemented in 8344962

**Review by darmiel — APPROVED** [review]

File: backend/api-documentation/user/modify user.bru [hunk 3]
```diff
@@ -0,0 +1,23 @@
+meta {
+  name: modify user
+  type: http
+  seq: 7
```
Comment (Julian702) [inline backend/api-documentation/user/modify user.bru:4]: Has to be 6 and /user/delete user.bru has to be updated to 7
```suggestion
  seq: 6
```

File: backend/api-documentation/user/modify user.bru [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+auth:bearer {
+  token: {{accessToken}}
+}
+
+body:json {
+  {
+    "userName": "",
+    "userEmail": "",
+    "userPassword":""
```
Comment (Julian702) [inline backend/api-documentation/user/modify user.bru:21]: Lets fill this with example values and swap email with name to match project default
```suggestion
    "userEmail": "test@test.test",
    "userName": "test",
    "userPassword":"test"
```

**Review by Julian702 — CHANGES_REQUESTED** [review]
Works fine, just some nits and inconsistent error messages compared to /api/register

File: backend/api-documentation/user/modify user.bru (hunk 3 above)
Comment (TomRomeo) [reply backend/api-documentation/user/modify user.bru]: changed in 74eee5b

File: backend/api-documentation/user/modify user.bru (hunk 4 above)
Comment (TomRomeo) [reply backend/api-documentation/user/modify user.bru]: changed in 74eee5b

File: backend/api-documentation/user/modify user.bru (hunk 3 above)
Comment (Julian702) [reply backend/api-documentation/user/modify user.bru]: You have not updated the documentation for the delete user endpoint. Although I even mentioned this^^

**Review by Julian702 — COMMENTED** [review]
You have implemented two code smells with your latest commit.

File: backend/api-documentation/user/modify user.bru [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+auth:bearer {
+  token: {{accessToken}}
+}
+
+body:json {
+  {
+    "userEmail": "test@test.test",
+    "userName": "test",
+    "userPassword":"test"
```
Comment (Julian702) [inline backend/api-documentation/user/modify user.bru:21]:
```suggestion
    "userPassword": "test"
```

**Review by Julian702 — CHANGES_REQUESTED** [review]
One last nit and it's also possible to change the password to "". After that I'm no longer able to login to my account.

**Review by Julian702 — APPROVED** [review]
Hast du dir verdient <3

**Pull request merged**